sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs

```diff
@@ -708,7 +710,45 @@ internal static LinkedEntityMatch LinkedEntityMatch(double confidenceScore = def
                 analyzeSentimentActionResults.ToList(),
                 recognizeCustomEntitiesActionResults.ToList(),
                 singleLabelClassifyActionResults.ToList(),
-                multiLabelClassifyActionResults.ToList()
+                multiLabelClassifyActionResults.ToList(),
+                new List<AnalyzeHealthcareEntitiesActionResult>()
+                );
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="TextAnalytics.AnalyzeActionsResult"/> for mocking purposes.
+        /// </summary>
+        /// <param name="extractKeyPhrasesActionResults">Sets the collection of <see cref="TextAnalytics.ExtractKeyPhrasesActionResult"/>.</param>
+        /// <param name="recognizeEntitiesActionResults">Sets the collection of <see cref="TextAnalytics.RecognizeEntitiesActionResult"/>.</param>
+        /// <param name="recognizePiiEntitiesActionResults">Sets the collection of <see cref="TextAnalytics.RecognizePiiEntitiesActionResult"/>.</param>
+        /// <param name="recognizeLinkedEntitiesActionResults">Sets the collection of <see cref="TextAnalytics.RecognizeLinkedEntitiesActionResult"/>.</param>
+        /// <param name="analyzeSentimentActionResults">Sets the collection of <see cref="TextAnalytics.AnalyzeSentimentActionResult"/>.</param>
+        /// <param name="recognizeCustomEntitiesActionResults">Sets the collection of <see cref="TextAnalytics.RecognizeCustomEntitiesActionResult"/>.</param>
+        /// <param name="singleLabelClassifyActionResults">Sets the collection of <see cref="TextAnalytics.SingleLabelClassifyActionResult"/>.</param>
+        /// <param name="multiLabelClassifyActionResults">Sets the collection of <see cref="TextAnalytics.MultiLabelClassifyActionResult"/>.</param>
+        /// <param name="analyzeHealthcareEntitiesActionResults">Sets the collection of <see cref="TextAnalytics.AnalyzeHealthcareEntitiesActionResult"/>.</param>
+        /// <returns>A new instance of <see cref="TextAnalytics.AnalyzeActionsResult"/> for mocking purposes.</returns>
+        public static AnalyzeActionsResult AnalyzeActionsResult(
+            IEnumerable<ExtractKeyPhrasesActionResult> extractKeyPhrasesActionResults,
```

Comment:
Add [EditorBrowsable(EditorBrowsableState.Never)]

Reply:
That only goes on the old ones that get overloaded, not the new ones added. The idea is that we won't break customers doing this, but don't want the older overloads to show up in IntelliSense anymore. See https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking-factory-builder-methods

Reply:
I thought we use ModelFactory class only for our own testing, do we expect customers to use it? If so, when?

Reply:
The model factory is a key part of our testability strategy.   Since we intentionally make output-only models non-constructable by normal means, the factory is what allows developers to create instances of them when mocking service methods.   

More information:
- [Blog article on unit testing the Azure SDK](https://devblogs.microsoft.com/azure-sdk/unit-testing-and-mocking/)
- [Azure.Core mocking sample](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Mocking.md) _(well, it demonstrates it but doesn't directly discuss it...)_